app.py

```
@@ -92,10 +99,68 @@ def main():
+            if f'zip_data_{keywords_str}' not in st.session_state:
+                import requests
+                import zipfile
+                import io
+                
+                with st.spinner("ZIP 파일 준비 중..."):
+                    # ZIP 파일 생성
+                    zip_buffer = io.BytesIO()
+                    with zipfile.ZipFile(zip_buffer, 'w', zipfile.ZIP_DEFLATED) as zip_file:
+                        for i, url in enumerate(all_image_urls, 1):
+                            try:
+                                response = requests.get(url)
```

Ruff (0.12.2)

Probable use of `requests` call without timeout

(S113)